File: PYTHON/PROGRAM/Baterai.py
# KELAS BATERAI
from Komponen import Komponen

class Baterai(Komponen):
    def __init__(self, merk="", nama="", produsen="", kapasitas=0, jenis_baterai="", siklus_charge=0, kesehatan_persen=0):
        Komponen.__init__(self, merk, nama, produsen)
        self.__kapasitas = kapasitas
        self.__jenis_baterai = jenis_baterai
        self.__siklus_charge = siklus_charge
        self.__kesehatan_persen = kesehatan_persen
    
    def set_kapasitas(self, kapasitas):
        self.__kapasitas = kapasitas
    
    def set_jenis_baterai(self, jenis_baterai):
        self.__jenis_baterai = jenis_baterai
    
    def set_siklus_charge(self, siklus_charge):
        self.__siklus_charge = siklus_charge
    
    def set_kesehatan_persen(self, kesehatan_persen):
        self.__kesehatan_persen = kesehatan_persen
    
    def get_kapasitas(self):
        return self.__kapasitas
    
    def get_jenis_baterai(self):
        return self.__jenis_baterai
    
    def get_siklus_charge(self):
        return self.__siklus_charge
    
    def get_kesehatan_persen(self):
        return self.__kesehatan_persen
    
    def tampilkan_info(self):
        Komponen.tampilkan_info(self)
        print("Kapasitas:", self.__kapasitas, "mAh")
        print("Jenis Baterai:", self.__jenis_baterai)
        print("Siklus Charge:", self.__siklus_charge)
        print("Kesehatan:", self.__kesehatan_persen, "%")

File: CPP/PROGRAM/Baterai.cpp
// KELAS BATERAI
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Baterai : public Komponen {
    private:
        int kapasitas; // mAh
        string jenisBaterai; // Li-Ion, Li-Po, Ni-Cd, dll
        int siklusCharge;
        int kesehatanPersen;

    public:
        Baterai() {}
        
        Baterai(string merk, string nama, string produsen, int kapasitas, string jenisBaterai, int siklusCharge, int kesehatanPersen) 
            : Komponen(merk, nama, produsen) {
            this->kapasitas = kapasitas;
            this->jenisBaterai = jenisBaterai;
            this->siklusCharge = siklusCharge;
            this->kesehatanPersen = kesehatanPersen;
        }
        
        void setKapasitas(int kapasitas) {
            this->kapasitas = kapasitas;
        }
        
        void setJenisBaterai(string jenisBaterai) {
            this->jenisBaterai = jenisBaterai;
        }
        
        void setSiklusCharge(int siklusCharge) {
            this->siklusCharge = siklusCharge;
        }
        
        void setKesehatanPersen(int kesehatanPersen) {
            this->kesehatanPersen = kesehatanPersen;
        }
        
        int getKapasitas() {
            return kapasitas;
        }
        
        string getJenisBaterai() {
            return jenisBaterai;
        }
        
        int getSiklusCharge() {
            return siklusCharge;
        }
        
        int getKesehatanPersen() {
            return kesehatanPersen;
        }
        
        void tampilkanInfo() {
            Komponen::tampilkanInfo();
            cout << "Kapasitas: " << kapasitas << " mAh" << endl;
            cout << "Jenis Baterai: " << jenisBaterai << endl;
            cout << "Siklus Charge: " << siklusCharge << endl;
            cout << "Kesehatan: " << kesehatanPersen << "%" << endl;
        }
        
    ~Baterai() {}
};

